src/devtools/client/debugger/src/components/Editor/Preview/NewObjectInspector.tsx

```diff
@@ -0,0 +1,43 @@
+import { Value as ProtocolValue } from "@replayio/protocol";
+import Inspector from "bvaughn-architecture-demo/components/inspector";
+import "bvaughn-architecture-demo/pages/inspector.css";
+import { preCacheObjects } from "bvaughn-architecture-demo/src/suspense/ObjectPreviews";
+import { clientValueToProtocolValue } from "bvaughn-architecture-demo/src/utils/protocol";
+import { ValueItem } from "devtools/packages/devtools-reps";
+import { ThreadFront } from "protocol/thread";
+import { useMemo } from "react";
+import { useAppSelector } from "ui/setup/hooks";
+
+import { getPreview } from "../../../selectors";
+
+export default function NewObjectInspector() {
+  const preview = useAppSelector(getPreview);
+  const pause = ThreadFront.currentPause;
+
+  // HACK
+  // The new Object Inspector does not consume ValueFronts.
+  // It works with the Replay protocol's Value objects directly.
+  // At the moment this means that we need to convert the ValueFront back to a protocol Value.
+  const protocolValue: ProtocolValue | null = useMemo(() => {
+    if (pause == null || pause.pauseId == null) {
+      return null;
+    } else if (preview == null || !preview.hasOwnProperty("root") == null) {
+      return null;
+    }
+
+    // TODO Move this into an action creator or whatever loads the Pause data.
+    preCacheObjects(pause.pauseId, Array.from(pause.objects.values()));
```

Comment:
I'm going to move this to a Thunk or something before I mark the PR as ready for review.

Reply:
I might be able to pass these objects through the `clientValueToProtocolValue` before passing them to `preCacheObjects` – which would let me get rid of the hack in `HTMLChildrenRenderer`.